bqn-mode.el

```diff
@@ -304,11 +305,28 @@ BQN buffers (or recreate them)."
   :type 'boolean
   :group 'bqn)
 
+(defun bqn--comint-prefix ()
+  "The prefix for BQNs comint buffers."
+  (concat "*" bqn-comint--process-name "-"))
+
+(defconst bqn--comint-suffix "*"
+  "The suffix for BQNs comint buffers.")
+
+(defun bqn--comint-buffer-name ()
+  "Return the name of the comint buffer associated to the current buffer.
+Note that the comint buffer may not exist yet, use `bqn-comint-buffer'
+to create it."
+  (let* ((pref (bqn--comint-prefix))
+         (buf (or (buffer-file-name) (buffer-name))))
+    (if (string-prefix-p pref buf)
+        buf
+      (concat pref buf bqn--comint-suffix))))
```

Comment:
One could also think of some internal state to keep track of comint buffer–buffer associations, but I figured since this is the only state that's currently needed this would be a bit overkill